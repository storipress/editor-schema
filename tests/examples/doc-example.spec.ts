import { describe, expect, it } from 'vitest'

// eslint-disable-next-line ts/no-require-imports
const { Node } = require('@tiptap/pm/model')

// eslint-disable-next-line ts/no-require-imports
const TipTapSchema = require('../../dist/editor-schema.cjs')

const { walkTree, schema, parse } = TipTapSchema

describe('parse', () => {
  it('parse to ProseMirror node', () => {
    const res = parse('<p>foo</p><img src="https://example.com" />')
    expect(res.toJSON()).toMatchSnapshot()
  })
})

describe('walkTree', () => {
  it('wrap image src', async () => {
    const frag = Node.fromJSON(schema, {
      type: 'doc',
      content: [
        {
          type: 'image',
          attrs: {
            src: 'https://example.com',
          },
        },
        {
          type: 'embed',
        },
        {
          type: 'paragraph',
          content: [
            {
              type: 'text',
              text: 'foo',
            },
          ],
        },
      ],
    })

    const res = await walkTree(frag, {
      image: (node) =>
        node.type.create({
          ...node.attrs,
          src: `$${node.attrs.src}$`,
        }),
    })

    expect(res.toJSON()).toMatchSnapshot()
  })
})
