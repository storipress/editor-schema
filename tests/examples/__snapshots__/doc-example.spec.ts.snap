// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parse > parse to ProseMirror node 1`] = `
{
  "content": [
    {
      "attrs": {
        "id": null,
      },
      "content": [
        {
          "text": "foo",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    {
      "attrs": {
        "alt": null,
        "cid": null,
        "file": null,
        "id": null,
        "link": "",
        "provider": null,
        "source": [],
        "src": "https://example.com",
        "style": null,
        "title": "",
        "type": "normal",
      },
      "type": "image",
    },
  ],
  "type": "doc",
}
`;

exports[`walkTree > wrap image src 1`] = `
{
  "content": [
    {
      "attrs": {
        "alt": "",
        "cid": null,
        "file": null,
        "id": null,
        "link": "",
        "provider": null,
        "source": [],
        "src": "$https://example.com$",
        "style": "",
        "title": "",
        "type": "regular",
      },
      "type": "image",
    },
    {
      "attrs": {
        "blockName": null,
        "content": "",
        "id": null,
        "name": null,
        "uuid": null,
      },
      "type": "embed",
    },
    {
      "attrs": {
        "id": null,
      },
      "content": [
        {
          "text": "foo",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
